Remotion/Core/UnitTests/Context/SafeContextTests/SafeContextTest.Task.ContinueWith.cs

```diff
@@ -0,0 +1,367 @@
+      {
+        throw new NotImplementedException();
+      }
+    }
+
+    private static readonly MethodInfo s_createTestInvocationMethod = typeof(SafeContextTaskTest).GetMethod(nameof(CreateTestInvocation), BindingFlags.NonPublic | BindingFlags.Static);
+
+    private readonly MethodInfo _runContinueWithTestMethod = typeof(SafeContextTaskTest).GetMethod(nameof(RunContinueWithTest), BindingFlags.NonPublic | BindingFlags.Instance);
+
+    [Test]
+    [TestCaseSource(nameof(ContinueWithTestSource), new object[] { true })]
+    public void ContinueWith_ImplicitSafeContextProvider (Type taskType, Type continuationType, Func<ISafeContextStorageProvider, object> continuationFactory, ContinueWithTestFlags flags)
+    {
+      var safeContextStorageProvider = SafeContextProviderMock.Object;
+      using (SetupImplicitSafeContextStorageProvider(safeContextStorageProvider))
+      {
+        var runTestMethod = _runContinueWithTestMethod.MakeGenericMethod(taskType, continuationType);
+        var continuation = continuationFactory(null);
+        InvokeWithInnerExceptionRethrow(runTestMethod, this, continuation, flags);
+      }
+    }
+
+    [Test]
+    [TestCaseSource(nameof(ContinueWithTestSource), new object[] { false })]
+    public void ContinueWith_ExplicitSafeContextProvider (Type taskType, Type continuationType, Func<ISafeContextStorageProvider, object> continuationFactory, ContinueWithTestFlags flags)
+    {
+      var safeContextStorageProvider = SafeContextProviderMock.Object;
+      var runTestMethod = _runContinueWithTestMethod.MakeGenericMethod(taskType, continuationType);
+      var continuation = continuationFactory(safeContextStorageProvider);
+      InvokeWithInnerExceptionRethrow(runTestMethod, this, continuation, flags);
+    }
+
+    /// <summary>
+    /// Converts the method overloads for <see cref="SafeContext"/>.<see cref="SafeContext.Task"/>.ContinueWith(...) to <see cref="TestCaseData"/>
+    /// that can be used to invoke the tests through <see cref="RunContinueWithTest{TTask,TContinuation}"/>.
+    /// </summary>
+    /// <seealso cref="ContinueWith_ImplicitSafeContextProvider"/>
+    /// <seealso cref="ContinueWith_ExplicitSafeContextProvider"/>
+    private static IEnumerable<TestCaseData> ContinueWithTestSource (bool implicitSafeContextProvider)
+    {
+      var methods = typeof(SafeContext.Task).GetMethods(BindingFlags.Public | BindingFlags.Static)
+          .Where(e => e.Name == "ContinueWith")
+          .ToArray();
+
+      foreach (var method in methods)
+      {
+        var parameters = method.GetParameters();
+        var isImplicit = parameters[0].ParameterType != typeof(ISafeContextStorageProvider);
+        if (isImplicit != implicitSafeContextProvider)
+          continue;
+
+        var taskType = ResolveGenericParameters(parameters[isImplicit ? 0 : 1].ParameterType);
+        Debug.Assert(typeof(Task).IsAssignableFrom(taskType), "typeof(Task).IsAssignableFrom(taskType)");
```

Comment:
Please use AssertionUtility.IsTrue/IsFalse, etc.

I avoid Debug.Assert in re-motion sources to not having to think about what causes/does-not-cause the message box to get displayed.

Reply:
Added a task in the backlog https://re-motion.atlassian.net/browse/RMTEAM-594